palm/containerizer.py

```diff
@@ -17,18 +17,18 @@ class Containerizer(ABC):
     Implementations of run() should call check_setup() and generate()
     """
 
-    def __init__(self, ctx, template_dir: Path) -> None:
+    def __init__(self, environment, template_dir: Path) -> None:
```

Comment:
This is actually a breaking change, since Containerizer is part of the public API and we're changing it's arguments. I believe palm-dbt uses this class directly and would need an update to continue working...

Reply:
Gotcha, so what do you think should we do? Revert? Or make a cascading update?

Reply:
https://github.com/palmetto/palm-dbt/blob/develop/palm/plugins/dbt/dbt_containerizer.py#L10

Subclassing PythonContainerizer... but not actually calling the parent constructor.

Still _technically_ a breaking change, even if the one external use I'm aware of may not actually break...

Reply:
The problem with a cascading change is that we could break plugins we don't know about (I think Ethan had one or two...). We probably want to revert this change and:
1. add an optional kwarg for `environment`
2. logic to use `environment` over `ctx`
3. Add a deprecation warning if `ctx` has an `obj`
4. update cmd_containerize to something like `PythonContainerizer({}, template_dir, version,environment=environment).run()`

There may be a cleaner way... I'll think about it more tonight.